### PR TITLE
LibGfx/JPEG: More debug output + Refactor two u8 in a struct

### PR DESCRIPTION
**Pull request by LucasChollet**

### CODE REVIEW

File: Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp [hunk 1]
```diff
@@ -973,6 +973,7 @@ static ErrorOr<void> read_restart_interval(JPEGStream& stream, JPEGLoadingContex
         return Error::from_string_literal("Malformed DRI marker found");
     }
     context.dc_restart_interval = TRY(stream.read_u16());
+    dbgln_if(JPEG_DEBUG, "Restart marker: {}", context.dc_restart_interval);
```
Comment (nico) [inline Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp:981]: Out if interest, what is this for? We don't use these for anything, and it's kinda noisy to dump.

**Review by nico — APPROVED** [review]
Lg

File: Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp [hunk 2]
```diff
@@ -1265,30 +1269,29 @@ static ErrorOr<void> read_start_of_frame(JPEGStream& stream, JPEGLoadingContext&
         component.index = i;
 
         u8 subsample_factors = TRY(stream.read_u8());
-        component.hsample_factor = subsample_factors >> 4;
-        component.vsample_factor = subsample_factors & 0x0F;
+        component.sampling_factors = SamplingFactors(subsample_factors >> 4, subsample_factors & 0x0F);
```
Comment (nico) [inline Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp:1272]: https://dev.azure.com/SerenityOS/SerenityOS/_build/results?buildId=42764&view=logs&jobId=2351e412-c793-598f-27f8-72c4f2d55ba1&j=2351e412-c793-598f-27f8-72c4f2d55ba1&t=9c50333a-ada2-5596-d44b-d8dcee270ad2

Probably have to use braced init here (?)

Comment (LucasChollet) [reply Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp:1272]: Brace init can't work as both value are `int` and not `u8`

File: Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp (hunk 1 above)
Comment (LucasChollet) [reply Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp:981]: `context.dc_restart_interval` is used in `decode_huffman_stream`
I'm not sure to understand you

File: Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp (hunk 2 above)
Comment (nico) [reply Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp]: Just have to add a cast (?)

**Pull request merged**